File: scala/Factorial.scala
import scala.io.StdIn.readInt

object Factorial extends App {
  println("Type the integer: ")
  val input = readInt()
  var total: Int = 1
  for (i <- 1 to input) {
    total = total * i
  }
  println(s"The factorial of $input is $total")
}

File: C++/Factorial.cpp
#include <iostream>
using namespace std;
 
int main()
{
	int n;
    long double factorial = 1.0;
	
	cout << "Input an integer:";
	cin >> n;
	if (n < 0)
		cout << "Error";
	 else 
        for(int i = 1; i <= n; ++i) 
            factorial *= i;	
	cout << "The factorial of number " << n << " is " << factorial << endl;
  
}